tests/unit/backend/corpora/dataset_processing/test_process.py

```diff
@@ -28,38 +39,127 @@ def test_extract_metadata(self, mock_read_h5ad):
+                "development_stage_ontology_term_id",
+            ],
             index=(str(i) for i in range(50001)),
         )
-        uns = {"organism": "Homo sapiens", "layer_descriptions": {"X": "raw"}}
+        uns = {
+            "organism": "Homo sapiens",
+            "organism_ontology_term_id": "NCBITaxon:8505",
+            "layer_descriptions": {"X": "raw"},
+        }
 
         adata = anndata.AnnData(X=df, obs=obs, uns=uns)
-
         mock_read_h5ad.return_value = adata
 
         extracted_metadata = process.extract_metadata("dummy")
+        lab, ont = "label", "ontology_term_id"
+
+        self.assertDictEqual(extracted_metadata["organism"], {lab: "Homo sapiens", ont: "NCBITaxon:8505"})
+
+        def list_equal(list1, list2, cmp_func):
+            self.assertEqual(len(list1), len(list2))
+            for el1 in list1:
+                self.assertIn(el1, list2)
+                el2 = list2[list2.index(el1)]
+                if cmp_func:
+                    cmp_func(el1, el2)
+
+        list_equal(
+            extracted_metadata["tissue"],
+            [{lab: "lung", ont: "UBERON:01"}, {lab: "liver", ont: "UBERON:10"}],
+            self.assertDictEqual,
+        )
+
+        list_equal(
+            extracted_metadata["assay"],
+            [{lab: "10x", ont: "EFO:001"}, {lab: "smartseq", ont: "EFO:010"}, {lab: "cite-seq", ont: "EFO:011"}],
+            self.assertDictEqual,
+        )
+
+        list_equal(extracted_metadata["disease"], [{lab: "healthy", ont: "MONDO:123"}], self.assertDictEqual)
 
-        self.assertEqual(extracted_metadata["organism"], "Homo sapiens")
-        self.assertListEqual(sorted(extracted_metadata["tissue"]), sorted(["lung", "liver"]))
-        self.assertListEqual(sorted(extracted_metadata["assay"]), sorted(["10x", "smartseq", "cite-seq"]))
-        self.assertListEqual(sorted(extracted_metadata["disease"]), sorted(["healthy"]))
         self.assertListEqual(sorted(extracted_metadata["sex"]), sorted(["male", "female"]))
-        self.assertListEqual(sorted(extracted_metadata["ethnicity"]), sorted(["solomon islander", "orcadian"]))
-        self.assertListEqual(sorted(extracted_metadata["development_stage"]), sorted(["adult", "baby", "tween"]))
+
+        list_equal(
+            extracted_metadata["ethnicity"],
+            [{lab: "solomon islander", ont: "HANCESTRO:321"}, {lab: "orcadian", ont: "HANCESTRO:456"}],
+            self.assertDictEqual,
+        )
+
+        list_equal(
+            extracted_metadata["development_stage"],
+            [{lab: "adult", ont: "HsapDv:0"}, {lab: "baby", ont: "HsapDv:1"}, {lab: "tween", ont: "HsapDv:2"}],
+            self.assertDictEqual,
+        )
         self.assertEqual(extracted_metadata["cell_count"], 50001)
         self.assertAlmostEqual(extracted_metadata["mean_genes_per_cell"], numpy.count_nonzero(df) / 50001)
 
+    def test_update_db(self):
+
+        collection = Collection.create(visibility=CollectionVisibility.PUBLIC)
```

Comment:
We should only update Private collections, but this check is handled earlier in the upload process. We should change to PRIVATE for consistency.